pkg/test/framework/components/istio/operator.go

```diff
@@ -215,6 +212,25 @@ func deploy(ctx resource.Context, env *kube.Environment, cfg Config) (Instance,
 			return nil, err
 		}
 	}
+	// If control plane cluster isn't the config cluster we are under external control plane mode which uses remote KUBECONFIG
```

Comment:
@irisdingbj I'm concerned that the code here is growing in complexity. I think we need to break this down into a more sensible flow so that we can understand it and maintain it going forward.  Here's what I propose:

This function will delegate to the following high-level methods that are responsible for part of the high-level flow:

- ConfigureTrust(): This is responsible for configuring trust as appropriate for all clusters in the mesh.
- InstallConfigClusters(): This pushes the Istio CRDs to all of the clusters that are to be used as sources of config. This is really only needed for "external control plane", but it's a sensible thing to do in general.
- InstallPrimaryClusters(): This will install the control plane on each cluster based on the controlPlaneTopology. For primaries that use other clusters for config, it will also set up the kubeconfigs.
- InstallRemoteClusters(): This will install the remote clusters (type depends on the value of the lightweight flag) based on the controlPlaneTopology.

This should help to make the workflow a little more clear. Within each of these functions, I'd like to do as little as possible that is specific to `external control planes`. I think the only special logic would be in `InstallPrimaryClusters` where it has to install the kubeconfig file.

WDYT?

FYI @stevenctl

Reply:
thinking we need to put this refactor into another separate PR, this change in this PR is big enough

Reply:
As I mentioned in the other thread, I think this PR needs to address this since it's adding complexity.